examples/selective2d/2d_train.py

```diff
@@ -0,0 +1,354 @@
+  parser.add_argument('--decay_lr', type=str_to_bool, default=True) # whether to decay the learning rate
+  parser.add_argument('--warmup_iters', type=int, default=2000)  
+  parser.add_argument('--lr_decay_iters', type=int, default=600000) 
+  parser.add_argument('--min_lr', type=float, default= 6e-5)  # minimum learning rate 
+  
+  # distributed
+  parser.add_argument('--backend', type=str, default="nccl")  # 'nccl', 'gloo', etc.
+  parser.add_argument('--compile', type=str_to_bool, default=False) # use PyTorch 2.0 to compile the model to be faster
+  parser.add_argument('--rank', type=int, default=int(os.environ["RANK"]))
+  parser.add_argument('--local_rank', type=int, default=int(os.environ["LOCAL_RANK"]))
+  parser.add_argument('--world_size', type=int, default=int(os.environ["WORLD_SIZE"]))
+  parser.add_argument('--device', type=str, default=f"cuda:{os.environ['LOCAL_RANK']}")
+  parser.add_argument('--master_process', type=str_to_bool, default=bool(os.environ['RANK']==0)) 
+  parser.add_argument('--tp_size', type=int, default=2)
+  parser.add_argument('--pp_size', type=int, default=2)
+
+  parser.add_argument('--debug', dest='debug', action='store_true')
+
+  args = parser.parse_args()
+
+  return args
+
+
+def rank_print(x):
+    _rank = os.getenv('RANK')
+    if _rank == '0':
+        print(x)
+
+def get_rand(args):
+    x = torch.randint(0, args.vocab_size, (args.batch_size, args.block_size), device=args.device)
+    y = torch.randint(0, args.vocab_size, (args.batch_size, args.block_size), device=args.device)
+    return x, y
+
+def tp_attention(model, name, mesh, tp_dim=0, q='q', k='k', v='v', o='c_proj'):
+  layer = model.get_submodule(name)
+  parallelize_module(layer, mesh, { q: ColwiseParallel(),
+                                    k: ColwiseParallel(),
+                                    v: ColwiseParallel(),
+                                    o: RowwiseParallel() },
+                                    tp_mesh_dim=tp_dim)
+
+  return model
+
+def tp_mlp(model, name, mesh, tp_dim=0, mlp='mlp'):
+  layer = model.get_submodule(name)
+  parallelize_module(layer, mesh, { mlp: PairwiseParallel() }, tp_mesh_dim=tp_dim)
+
+  return model
+
+
+def tp(model, n_layer, mesh, offset=0, tp_dim=0):
+  for i in range(n_layer):
+    block = model.get_submodule(f'transformer.h.{i + offset}')
+    parallelize_module(block, mesh, {'attn.q': ColwiseParallel(),
+                                     'attn.k': ColwiseParallel(),
+                                     'attn.v': ColwiseParallel(),
+                                     'attn.c_proj': RowwiseParallel(),
+                                     'mlp': PairwiseParallel()},
+                                     tp_mesh_dim=tp_dim) 
+
+  return model
+
+def pp(model, pp_device_mesh, args):
+  from pippy.IR import annotate_split_points, PipeSplitWrapper 
+  from pippy import split_into_equal_size
```

Comment:
Any reason why this is imported but not used?

Reply:
My bad. I used in my draft and moved on to my own cut. Will remove.